File: cpp/2460.cpp
class Solution {
public:
    vector<int> applyOperations(vector<int>& nums) {
        int n = nums.size() - 1;
        vector<int> res(n + 1);
        int s = 0, e = n, i = 0;
        while (i <= n)
            if (i < n && nums[i] != 0 && nums[i] == nums[i + 1]) {
                res[s++] = nums[i] * 2;
                res[e--] = 0;
                i += 2;
            } else {
                if (nums[i] == 0)
                    res[e--] = 0;
                else
                    res[s++] = nums[i];
                i++;
            }
        return res;
    }
};

File: python3/2460.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        n = len(nums) - 1
        res = [0] * (n + 1)
        s = 0
        i = 0
        while i <= n:
            if i < n and nums[i] != 0 and nums[i] == nums[i + 1]:
                res[s] = nums[i] * 2
                s += 1
                i += 2
            else:
                if nums[i] != 0:
                    res[s] = nums[i]
                    s += 1
                i += 1
        return res
